agent/exec/container/controller.go

```diff
@@ -194,6 +195,12 @@ func (r *controller) Wait(pctx context.Context) error {
 				// If we get here, something has gone wrong but we want to exit
 				// and report anyways.
 				return ErrContainerDestroyed
+			case "health_status: unhealthy":
```

Comment:
I think it's more like a subscription mechanism: once engine marks the container as unhealthy, it sends this event to ones that listen on the eventq. And for engine, it doesn't need to poll health status periodically, (or we can say engine's health check mechanism already updates health status periodically).